homeassistant/bootstrap.py

```diff
@@ -890,69 +882,47 @@ async def _async_set_up_integrations(
     domains_to_setup, integration_cache = await _async_resolve_domains_to_setup(
         hass, config
     )
+    stage_2_domains = domains_to_setup.copy()
 
     # Initialize recorder
     if "recorder" in domains_to_setup:
         recorder.async_initialize_recorder(hass)
 
-    pre_stage_domains = [
-        (name, domains_to_setup & domain_group) for name, domain_group in SETUP_ORDER
+    stage_0_and_1_domains: list[tuple[str, set[str], int | None]] = [
+        *(
+            (name, domain_group & domains_to_setup, None)
```

Comment:
Why don't we have a timeout for these domains? If there's a hang on the initial groups, I think we should detect that too and give up?